diff --git a/config/win/BUILD.gn b/config/win/BUILD.gn
index a5075393b..3e282efd0 100644
--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -86,6 +86,8 @@ config("compiler") {
   # modes.
   cflags_c = [ "/TC" ]
   cflags_cc = [ "/TP" ]
+  
+  cflags_cc += [ "/std:c++17" ]
 
   cflags += [
     # Work around crbug.com/526851, bug in VS 2015 RTM compiler.
@@ -489,7 +491,7 @@ config("release_crt") {
       ldflags = [ "/DEFAULTLIB:msvcprt.lib" ]
     }
   } else {
-    cflags = [ "/MT" ]
+    cflags = [ "/MD" ]
     if (use_custom_libcxx) {
       ldflags = [ "/DEFAULTLIB:libcpmt.lib" ]
     }
@@ -514,12 +516,12 @@ config("dynamic_crt") {
 config("static_crt") {
   if (is_debug) {
     # This pulls in the static debug CRT and defines _DEBUG
-    cflags = [ "/MTd" ]
+    cflags = [ "/MDd" ]
     if (use_custom_libcxx) {
       ldflags = [ "/DEFAULTLIB:libcpmtd.lib" ]
     }
   } else {
-    cflags = [ "/MT" ]
+    cflags = [ "/MD" ]
     if (use_custom_libcxx) {
       ldflags = [ "/DEFAULTLIB:libcpmt.lib" ]
     }
